inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/activation/activation_kernel_opt.cpp

```diff
@@ -27,12 +27,35 @@ ParamsKey ActivationKernelOpt::GetSupportedKey() const {
     return k;
 }
 
+static size_t GetTotalSize(const activation_params& params) {
+    const auto input = params.inputs[0];
+    size_t totalSize = input.LogicalSize();
+    switch (params.inputs[0].GetLayout()) {
+        case DataLayout::b_fs_yx_fsv4:
+            totalSize = (totalSize / input.Feature().v) * Align(input.Feature().v, 4);
+            break;
+        case DataLayout::b_fs_yx_fsv16:
+        case DataLayout::b_fs_zyx_fsv16:
+            totalSize = (totalSize / input.Feature().v) * Align(input.Feature().v, 16);
+            break;
+        case DataLayout::b_fs_yx_fsv32:
+        case DataLayout::b_fs_zyx_fsv32:
+        case DataLayout::fs_b_yx_fsv32:
+            totalSize = (totalSize / input.Feature().v) * Align(input.Feature().v, 32);
+            break;
+        case DataLayout::bs_fs_zyx_bsv16_fsv16:
+        case DataLayout::bs_fs_yx_bsv16_fsv16:
+            totalSize = (totalSize / (input.Feature().v * input.Batch().v)) * Align(input.Feature().v, 16) * Align(input.Batch().v, 16);
```

Comment:
What will happen if `params.inputs[0].GetLayout()` value is not processed by this switch (e.g. we add new tensor format)? The kernel accepts all input layouts, but seems like the it'll work incorrectly, won't it?

Reply:
Yes. I think we need an API in the DataTensor to get the physical size. However there is not.. so I added this for a quick fix.

Reply:
I think I need some exception in the default case below..

Reply:
What about the following method?
```
    // Whole buffer size (in elements)
    size_t PhysicalSize() const { return totalSize; }

    // Whole buffer size (in bytes)
    size_t PhysicalSizeInBytes() const { return totalSize * ElementSize(); }
```

Reply:
As I checked, PhysicalSize () is defined in TensorBase wehre no blocked format info is knwon. So it does not calculate the padding added due to the blocked format.. 
Or I think we can change this activation_opt kernel to be used only for non_blocked formats. I think most of the cases, activations are fused to the previous op not being uses as a separate primitive, so I think it might not affect the overall performance...